app/models/namespace.rb

```diff
@@ -48,6 +49,22 @@ def self.get_from_name(name)
     [namespace, name]
   end
 
+  # Find the namespace for the given push event.
+  def self.find_from_event(event)
```

Comment:
This method is almost identical to the `Namespace.get_from_name` one. Are you sure that you can't re-use the code from the other method ? (either in the caller, or by adding an extra parameter in `Namespace.get_from_name` or something).